src/sentry/issues/endpoints/organization_group_search_views.py

```diff
@@ -72,3 +76,70 @@ def get(self, request: Request, organization: Organization) -> Response:
             order_by="position",
             on_results=lambda x: serialize(x, request.user, serializer=GroupSearchViewSerializer()),
         )
+
+    def put(self, request: Request, organization: Organization) -> Response:
+        if not features.has("organizations:issue-stream-custom-views", organization):
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        serializer = GroupSearchViewRestSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        validated_data = serializer.validated_data
+        try:
+            with transaction.atomic(using=router.db_for_write(GroupSearchView)):
+                bulk_update_views(organization, request.user.id, validated_data)
+
+                query = GroupSearchView.objects.filter(
+                    organization=organization, user_id=request.user.id
+                )
+        except IntegrityError:
```

Comment:
should we capture a sentry error here? seems like right now it would silently fail